CASMINST-6830 1.5.1 Install Upgrade Docs (#4888)
Cray-HPE/docs-csm: upgrade/1.5.1/README.md

* Fixed an issue where `cm health report slingshot refresh` command was giving a trace back error
* Updated documentation for IUF to include rolling reboots after upgrading [HSN](../../glossary.md#high-speed-network-hsn) NIC firmware
* Updated documentation to provide instructions for creating a new Nexus repository and adding RPMs
* Updated documentation for [CAPMC](../../glossary.md#cray-advanced-platform-monitoring-and-control-capmc) to warn about URL character limits
* Fixed an issue where BOS v2 requests for too many nodes from PCS exceeding the URL character limit
* Updated Nexus export and restore script to add better checks for determining existence of `nexus-bak` PVC or `nexus-backup` job
* Updated documentation for `fix_failed_to_start_etcd_on_master` to better specify how to add a process
* Fixed an issue with IUF where process-media/pre-install-check dislikes PDF file
* Fixed an issue where `backup_smd_postgres.sh` script is not executable
* `CAST-34869`: Fixed an issue where `cray-hms-rts-snmp` would enter encounter a segmentation fault
* `CAST-34141`: Fixed an issue where the `QLogic` driver would crash the system

> 

Reviewer: Add a note about fix for spire-agent version issue fixed with hotfix `CASMTRIAGE-6715-noos-repo`
```suggestion
* Fixed an issue where an old version of the spire-agent was loaded into Nexus resulting in failed ARM compute builds

```

## Steps

1. [Preparation](#preparation)
1. [Setup Nexus](#setup-nexus)

> 1. [Update Argo CRDs](#update-argo-crds)

Reviewer 2: ```suggestion
```

1. [Upgrade services](#upgrade-services)
1. [Upload NCN images](#upload-ncn-images)

> 1. [Upgrade Ceph and stop local Docker registries](#upgrade-ceph-and-stop-local-docker-registries)

Reviewer 3: ```suggestion
```

> 1. [Enable `smartmon` metrics on storage NCNs](#enable-smartmon-metrics-on-storage-ncns)

Reviewer 3: ```suggestion
```

   export CSM_DISTDIR="$(pwd)/csm-1.5.1"
   echo "${CSM_DISTDIR}"
   ```

1. (`ncn-m001#`) Set `CSM_RELEASE_VERSION` to the CSM release version.

   ```bash
   export CSM_RELEASE_VERSION="$(${CSM_DISTDIR}/lib/version.sh --version)"
   echo "${CSM_RELEASE_VERSION}"
   ```

1. Download and install/upgrade the **latest** documentation on `ncn-m001`.

   See [Check for Latest Documentation](../../update_product_stream/README.md#check-for-latest-documentation).

### Setup Nexus

(`ncn-m001#`) Run `lib/setup-nexus.sh` to configure Nexus and upload new CSM RPM repositories, container images, and
Helm charts:

```bash
cd "$CSM_DISTDIR"
./lib/setup-nexus.sh ; echo "RC=$?"
```

On success, the output should end with the following:

```text
+ Nexus setup complete
setup-nexus.sh: OK
RC=0
```

In the event of an error,
consult [Troubleshoot Nexus](../../operations/package_repository_management/Troubleshoot_Nexus.md)
to resolve potential problems and then try running `setup-nexus.sh` again. Note that subsequent runs of `setup-nexus.sh`
may report `FAIL` when uploading duplicate assets. This is okay as long as `setup-nexus.sh` outputs `setup-nexus.sh: OK`
and exits with status code `0`.

> ### Update Argo CRDs
> 
> (`ncn-m001#`) Run the following script in preparation for 1.5.1 patch upgrade:
> 
> ```bash
> for c in $(kubectl get crd |grep argo | cut -d' ' -f1)
> do
>    kubectl label --overwrite crd $c app.kubernetes.io/managed-by="Helm"
>    kubectl annotate --overwrite crd $c meta.helm.sh/release-name="cray-nls"
>    kubectl annotate --overwrite crd $c meta.helm.sh/release-namespace="argo"
> done
> ```
> 

Reviewer 2: ```suggestion
```
Remove the entire "Update Argo CRDs" block as it's fixed in CSM 1.5 and doesn't need to be in the patch instructions

### Upgrade services

(`ncn-m001#`) Run `upgrade.sh` to deploy upgraded CSM applications and services:

```bash
cd "$CSM_DISTDIR"
./upgrade.sh
```

### Upload NCN images

It is important to upload NCN images to IMS and to edit the `cray-product-catalog`. This is necessary when updating
products with IUF. If this step is skipped, IUF will fail when updating or upgrading products in the future.

(`ncn-m001#`) Execute script to upload CSM NCN images and update the `cray-product-catalog`.

```bash
/usr/share/doc/csm/upgrade/scripts/upgrade/upload-ncn-images.sh
```

> ### Upgrade Ceph and stop local Docker registries
> 
> **Note:** This step may not be necessary if it was already completed by the CSM `v1.3.5` patch or `v1.4.2` patch. If it was already run,
> the following steps can be re-executed to verify that Ceph daemons are using images in Nexus and the local Docker
> registries have been stopped.
> 
> These steps will upgrade Ceph to `v16.2.13`. Then the Ceph monitoring daemons' images will be pushed to Nexus and the
> monitoring daemons will be redeployed so that they use these images in Nexus. Once this is complete, all Ceph daemons
> should be using images in Nexus and not images hosted in the local Docker registry on storage nodes. The third step
> stops the local Docker registry on all storage nodes.
> 
> 1. (`ncn-m001#`) Run Ceph upgrade to `v16.2.13`.
> 
>    ```bash
>    /usr/share/doc/csm/upgrade/scripts/ceph/ceph-upgrade-tool.py --version "v16.2.13"
>    ```
> 
> 1. (`ncn-m001#`) Redeploy Ceph monitoring daemons so they are using images in Nexus.
> 
>    ```bash
>    scp /usr/share/doc/csm/scripts/operations/ceph/redeploy_monitoring_stack_to_nexus.sh ncn-s001:/srv/cray/scripts/common/redeploy_monitoring_stack_to_nexus.sh
>    ssh ncn-s001 "/srv/cray/scripts/common/redeploy_monitoring_stack_to_nexus.sh"
>    ```
> 
> 1. (`ncn-m001#`) Stop the local Docker registries on all storage nodes.
> 
>    ```bash
>    scp /usr/share/doc/csm/scripts/operations/ceph/disable_local_registry.sh ncn-s001:/srv/cray/scripts/common/disable_local_registry.sh
>    ssh ncn-s001 "/srv/cray/scripts/common/disable_local_registry.sh"
>    ```
> 
> ### Enable `smartmon` metrics on storage NCNs
> 
> This step will install the `smart-mon` rpm on storage nodes, and reconfigure the `node-exporter` to provide `smartmon`
> metrics.
> 
> 1. (`ncn-m001#`) Execute the following script.
> 
>    ```bash
>    /usr/share/doc/csm/scripts/operations/ceph/enable-smart-mon-storage-nodes.sh
>    ```
> 

Reviewer 3: ```suggestion
```

Remove the entire blocks 'upgrade ceph and stop local docker registries' and 'Enable smartmon on storage nodes'. Both of these are fixed in CSM 1.5 and don't need to be in the patch instructions

# CSM 1.5.1 Patch Installation Instructions

* [Introduction](#introduction)
* [Bug fixes and improvements](#bug-fixes-and-improvements)
* [Steps](#steps)

## Introduction

This document guides an administrator through the patch update to Cray Systems Management `v1.5.1` from `1.5.0`.
If upgrading from CSM `v1.4.x` directly to `v1.5.1`, follow the procedures described
in [Upgrade CSM](../README.md) instead.

## Bug fixes and improvements

* Updated [IMS](../../glossary.md#image-management-service-ims) to allow for remote node builds

> * Support for Paradise hardware

Reviewer 4: Need to remove this from 1.5.1 docs.   Paradise support will be in 1.5.2

# CSM 1.5.1 Patch Installation Instructions

* [Introduction](#introduction)
* [Bug fixes and improvements](#bug-fixes-and-improvements)
* [Steps](#steps)

## Introduction

This document guides an administrator through the patch update to Cray Systems Management `v1.5.1` from `1.5.0`.
If upgrading from CSM `v1.4.x` directly to `v1.5.1`, follow the procedures described
in [Upgrade CSM](../README.md) instead.

## Bug fixes and improvements

> * Updated IMS to allow for remote node builds

Reviewer 5: ```suggestion
* Updated [IMS](../../glossary.md#image-management-service-ims) to allow for remote node builds
```

* Support for Paradise hardware
* Fixed an issue where `hms-discovery` would put default credentials into vault disabling non default credentials
* Fixed an issue where `cray-ipxe` generated scripts would cause boot errors

> * Fixed an issue where `BOS v2` would send large queries to CFS resulting 503 and 431 responses

Reviewer 5: ```suggestion
* Fixed an issue where [BOS](../../glossary.md#boot-orchestration-service-bos) v2 would send large queries to [CFS](../../glossary.md#configuration-framework-service-cfs) resulting 503 and 431 responses
```

> * Fixed an issue where PCS was adding invalid components to power operations

Reviewer 5: ```suggestion
* Fixed an issue where [PCS](../../glossary.md#power-control-service-pcs) was adding invalid components to power operations
```

> * Fixed an issue where IUF would encounter a race condition and stall on transitioning to new stages

Reviewer 5: ```suggestion
* Fixed an issue where [IUF](../../glossary.md#install-and-upgrade-framework-iuf) would encounter a race condition and stall on transitioning to new stages
```

* Fixed an issue where `Thanos` service is configured without storage limits
* Fixed an issue where `cray-dns-unbound-manager` `stderr` handling can corrupt configuration

> * Fixed an issue where "sat status" unnecessarily queries BOS for session template for every component

Reviewer 5: ```suggestion
* Fixed an issue where [SAT](../../glossary.md#system-admin-toolkit-sat) `status` unnecessarily queries BOS for session template for every component
```

> * Fixed an issue where a PATCH to a BOS v2 session to change it's name results in a bad state

Reviewer 5: ```suggestion
* Fixed an issue where a PATCH to a BOS v2 session to change its name results in a bad state
```

* Updated `cray-hms-rts-init` job to include a TTL

> * Updated node-exporter config to monitor `snmp` counters

Reviewer 5: ```suggestion
* Updated `node-exporter` configuration to monitor SNMP counters
```

> * Updated docs to cover switch configuration for NCNs

Reviewer 5: ```suggestion
* Updated documentation to cover switch configuration for [NCNs](../../glossary.md#non-compute-node-ncn)
```

* Fixed an issue where a PATCH call to BOS v2 components with a filter and non existent component ID would result in 503 and 431 errors
* Fixed an issue where BOS operators would output errors when all nodes exceed retry limit
* Fixed an issue where `cray-upload-recovery-images` fails to upload recovery firmware

> * Fixed an issue where "cm health report slingshot refresh" `cmd` was giving a trace back error

Reviewer 5: ```suggestion
* Fixed an issue where `cm health report slingshot refresh` command was giving a trace back error
```

> * Updated docs for IUF to include rolling reboots after upgrading HSN NIC firmware
> * Updated docs to provide instructions for creating a new Nexus repo and adding `rpms`
> * Updated docs for CAPMC to warn about URL character limits
> * Fixed an issue where BOS v2 requests for too many nodes from PCS exceeding the URL character limit
> * Updated docs script `nexus_export_and_restore` to add better checks for determining existence of `nexus-bak` PVC or nexus-backup job
> * Updated docs for `fix_failed_to_start_etcd_on_master` to better specify how to add a process

Reviewer 5: ```suggestion
* Updated documentation for IUF to include rolling reboots after upgrading [HSN](../../glossary.md#high-speed-network-hsn) NIC firmware
* Updated documentation to provide instructions for creating a new Nexus repository and adding RPMs
* Updated documentation for [CAPMC](../../glossary.md#cray-advanced-platform-monitoring-and-control-capmc) to warn about URL character limits
* Fixed an issue where BOS v2 requests for too many nodes from PCS exceeding the URL character limit
* Updated Nexus export and restore script to add better checks for determining existence of `nexus-bak` PVC or `nexus-backup` job
* Updated documentation for `fix_failed_to_start_etcd_on_master` to better specify how to add a process
```

* Fixed an issue with IUF where process-media/pre-install-check dislikes PDF file
* Fixed an issue where `backup_smd_postgres.sh` script is not executable
* `CAST-34869`: Fixed an issue where `cray-hms-rts-snmp` would enter encounter a segmentation fault
* `CAST-34141`: Fixed an issue where the `QLogic` driver would crash the system

## Steps

> 1. [CSM 1.5.1 Patch Installation Instructions](#csm-151-patch-installation-instructions)
> 1. [Introduction](#introduction)
> 1. [Bug fixes and improvements](#bug-fixes-and-improvements)
> 1. [Steps](#steps)

Reviewer 5: ```suggestion
```

   ssh ncn-s001 "/srv/cray/scripts/common/disable_local_registry.sh"
   ```

### Enable `smartmon` metrics on storage NCNs

This step will install the `smart-mon` rpm on storage nodes, and reconfigure the `node-exporter` to provide `smartmon`
metrics.

1. (`ncn-m001#`) Execute the following script.

   ```bash
   /usr/share/doc/csm/scripts/operations/ceph/enable-smart-mon-storage-nodes.sh
   ```

### Update management node CFS configuration

This step updates the CFS configuration which is set as the desired configuration for the management
nodes (NCNs). It ensures that the CFS configuration layers reference the correct commit hash for the
version of CSM being installed. It then waits for the components to reach a configured state in CFS.

1. (`ncn-m001#`)

   ```bash
   cd "$CSM_DISTDIR"
   ./update-mgmt-ncn-cfs-config.sh --base-query role=management \
      --save --create-backups --clear-error
   ```

   The output will look similar to the truncated output shown below.

   ```text
   INFO: Querying CFS configurations for the following NCNs: x3000c0s5b0n0, ...

>    INFO: Found configuration "management-csm-1.4.0" for component x3000c0s5b0n0
>    ...
>    INFO: Updating existing layer with repo path /vcs/cray/csm-config-management.git and playbook site.yml
>    INFO: Property "commit" of layer with repo path /vcs/cray/csm-config-management.git and playbook site.yml updated ...
>    INFO: Property "name" of layer with repo path /vcs/cray/csm-config-management.git and playbook site.yml updated ...
>    INFO: No layer with repo path /vcs/cray/csm-config-management.git and playbook ncn-initrd.yml found.
>    INFO: Adding a layer with repo path /vcs/cray/csm-config-management.git and playbook ncn-initrd.yml to the end.
>    INFO: Successfully saved CFS configuration "management-csm-1.4.0-backup-20230918T205149"
>    INFO: Successfully saved CFS configuration "management-csm-1.4.0"

Reviewer 5: Ideally we should just get actual output from an upgrade, but failing that, this makes the changes that I can anticipate:

```suggestion
   INFO: Found configuration "management-csm-1.5.0" for component x3000c0s5b0n0
   ...
   INFO: Updating existing layer with repo path /vcs/cray/csm-config-management.git and playbook ncn_nodes.yml
   INFO: Property "commit" of layer with repo path /vcs/cray/csm-config-management.git and playbook ncn_nodes.yml updated ...
   INFO: Property "name" of layer with repo path /vcs/cray/csm-config-management.git and playbook ncn_nodes.yml updated ...
   INFO: No layer with repo path /vcs/cray/csm-config-management.git and playbook ncn-initrd.yml found.
   INFO: Adding a layer with repo path /vcs/cray/csm-config-management.git and playbook ncn-initrd.yml to the end.
   INFO: Successfully saved CFS configuration "management-csm-1.5.0-backup-20240410T205149"
   INFO: Successfully saved CFS configuration "management-csm-1.5.0"
```
